### PR TITLE
Fix kicad-data.cabal so that it builds

### PR DESCRIPTION
**Pull request by mignon-p**
I attempted to build kicad-data, and I ran into a couple of problems:

First, building kicad-data from Hackage does not work, because `Data/Kicad/Util.hs` is missing from the tarball.  In order for it to be included, it needs to be listed in `kicad-data.cabal`.

Second, the lower bound for parsec is too loose, because `endOfLine` did not appear in parsec until version 3.1.6.

This pull request fixes both of these issues.  Thanks!

### CODE REVIEW

**Comment by kasbah** [comment]
Thanks for reporting and fixing ppelleti, I am curious what you are using this lib for? 

I am surprised the first issue doesn't come up when running the tests :confused: 

FYI I have two issue with the CI build, so don't worry too much about the build failure: 

1. KiCad broke the python API used to check whether the output is valid, they will fix this soon
2. `die_length` pad properties started appearing in footprints out in the wild and I haven't implemented these. Technically they shouldn't be in footprints but only in PCB files so I need to think about how to best handle them (there are a few other PCB only attributes this lib already parses, but it's not really the _correct_ approach).

**Comment by mignon-p** [comment]
> I am curious what you are using this lib for?

I'm just experimenting right now, but I have a few ideas in mind.  One is to generate or modify footprints programmatically.

Another thing I'd like to write is a program to take the design rules from one `.kicad_pcb` file and copy them into a different `.kicad_pcb` file.  Although your package doesn't support PCB files, I should be able to use the S-expression parser.

I'd been just about to write my own S-expression parser when I discovered yours.

In short, there isn't any grand plan, but I just want to mess around with manipulating Kicad files programmatically, and Haskell is my preferred language.

**Comment by kasbah** [comment]
Cool stuff, you might also be interested in this editor I am playing with which will let you embed Racket code into the KiCad s-expressions: [monostable/footwork](https://github.com/monostable/footwork) (not usable yet though).

My goal with kicad-data is to extend it to PCBs eventually so if you make any progress in that regard definitely send a pull-request!

This project was my first Haskell project and my first parser so there is some awkward code in places. For instance: I don't think the s-expression parser needs to be aware of the KiCad "keywords" and a lot of what's in PcbnewExpr parser should be re-written using `fold`s. I plan to clean that up eventually.

I have put up all [known issues](https://github.com/monostable/haskell-kicad-data/issues) now so if you come across any others, definitely let me know!
